warehouse/scripts/dbt_artifacts/__init__.py

```diff
@@ -193,119 +137,118 @@ def gvattrs(self) -> Dict[str, Any]:
         }
 
 
-class Seed(BaseNode):
-    resource_type: Literal[DbtResourceType.seed]
-
-    @property
-    def gvattrs(self) -> Dict[str, Any]:
-        return {
-            "fillcolor": "green",
-        }
-
+class AnalysisNode(BaseAnalysisNode, NodeModelMixin):
+    pass
 
-class Source(BaseNode):
-    resource_type: Literal[DbtResourceType.source]
 
-    @property
-    def gvattrs(self) -> Dict[str, Any]:
-        return {
-            "fillcolor": "blue",
-        }
+class SingularTestNode(BaseSingularTestNode, NodeModelMixin):
+    pass
 
 
-# TODO: this should be a discriminated type based on materialization
-class Model(BaseNode):
-    resource_type: Literal[DbtResourceType.model]
-    depends_on: NodeDeps
+class HookNode(BaseHookNode, NodeModelMixin):
+    pass
 
-    @property
-    def children(self) -> List["Model"]:
-        children = []
-        for unique_id, node in BaseNode._instances.items():
-            if (
-                isinstance(node, Model)
-                and node.depends_on.nodes
-                and self.unique_id in node.depends_on.nodes
-            ):
-                children.append(node)
-        return children
 
+class ModelNode(BaseModelNode, NodeModelMixin):
     @property
     def gvrepr(self) -> str:
         if (
-            self.config.materialized
-            in (DbtMaterializationType.table, DbtMaterializationType.incremental)
+            self.config
+            and self.config.materialized in ("table", "incremental")
             and self.catalog_entry
-            and self.catalog_entry.num_bytes
+            and self.catalog_entry.num_bytes  # type: ignore[attr-defined]
         ):
             return "\n".join(
                 [
-                    super(Model, self).gvrepr,
-                    f"Storage: {humanize.naturalsize(self.catalog_entry.num_bytes)}",
+                    super(ModelNode, self).gvrepr,
+                    f"Storage: {humanize.naturalsize(self.catalog_entry.num_bytes)}",  # type: ignore[attr-defined]
                 ]
             )
-        return super(Model, self).gvrepr
+        return super(ModelNode, self).gvrepr
 
     @property
     def gvattrs(self) -> Dict[str, Any]:
         fillcolor = "white"
 
-        if self.config.materialized in (
-            DbtMaterializationType.table,
-            DbtMaterializationType.incremental,
-        ):
+        if self.config and self.config.materialized in ("table", "incremental"):
             fillcolor = "aquamarine"
```

Comment:
It's turned off currently but it will still run locally. There's actually a solution described in https://github.com/iterative/cml/issues/1377 that I can implement soon-ish.

Reply:
follow-up PR here https://github.com/cal-itp/data-infra/pull/2605